### ISSUE TITLE
Building 2.24.2 for aarch64-apple-darwin with 11.0 SDK fails

### ISSUE THREAD

**Issue by giordano** [libsdl-org/SDL#6491]
When building SDL2 v2.24.2 for aarch64-apple-darwin with the 11.0 SDK I get a linking error:
```
Undefined symbols for architecture arm64:
  "___isPlatformVersionAtLeast", referenced from:
      _IOS_AddJoystickDevice in SDL_mfijoystick.m.o
```
This looks a problem similar to #4958.  Commenting out https://github.com/libsdl-org/SDL/blob/55b03c7493a7abed33cf803d1380a40fa8af903f/src/joystick/iphoneos/SDL_mfijoystick.m#L229-L233 I can build successfully.

**Comment by sezero** [comment]
Links fine for me when cross-compiling from linux (clang-12.0.1, against 11.3 sdk). Your clang probably can't locate `libclang_rt.osx.a`?

**Comment by icculus** [comment]
@giordano How are you building SDL? With Xcode? CMake? The configure script?

**Comment by giordano** [comment]
> Links fine for me when cross-compiling from linux (clang-12.0.1, against 11.3 sdk). Your clang probably can't locate `libclang_rt.osx.a`?

Linking to `libclang_rt.osx.a` did the trick, thanks!

> @giordano How are you building SDL? With Xcode? CMake? The configure script?

I'm cross-building with [`BinaryBuilder.jl`](https://binarybuilder.org/).  I was using the `configure` script (can't switch to CMake because the shared library on macOS would have a different soname, breaking the ABI, that's probably a different issue).

**Comment by sezero** [comment]
> > Links fine for me when cross-compiling from linux (clang-12.0.1, against 11.3 sdk). Your clang probably can't locate `libclang_rt.osx.a`?
> 
> Linking to `libclang_rt.osx.a` did the trick, thanks!

clang should have did that automatically, but all is good I guess now that you worked around it :)

**Comment by giordano** [comment]
Eh, ideally in BinaryBuilder we'd like to have the runtime libraries for the target platforms always available, but at the moment those are in a separate package that needs to be installed specifically when needed, which is what I did for libsdl2.  I'd never realised `___isPlatformVersionAtLeast` is implemented in `libclang_rt.osx`, I bump into that missing symbol every now and then, so that's quite useful to know!

Side note, `LIBS` doesn't seem to be picked when linking `libsdl2`, I had to put `-lclang_rt.osx` in `LDFLAGS` (I'm using `configure`, as I mentioned above).
